edx_exams/apps/lti/views.py

```diff
@@ -6,338 +6,91 @@
-    # TODO: The xblock-lti-consumer library does not currently support setting these attributes.
-    resource_link = get_resource_link()
-
-    lti_consumer.enable_proctoring(
-        # NOTE TO SELF: attempt_number is an auto-incrementing integer from 1 per learner, per assessment.
-        29,  # attempt_number,
-        session_data,
-        resource_link,
-        start_assessment_url=start_assessment_url,
-    )
-
-    # This is necessary for testing with the IMS tool, since the user will be an AnonymousUser without an id.
-    # TODO: Replace this with the authenticated user's id.
-    # TODO: Remove this once testing is complete.
-    user_id = 1 if request.user.id is None else request.user.id
+    # TODO: Here we'd do all the end of assessment things.
 
-    # Required user claim data
-    lti_consumer.set_user_data(
-        user_id=user_id,
-        # Pass Django user role to library
-        # TODO: A role of 'student' is not correctly mapped to the corresponding LTI claim for the Proctoring
-        #       Specification.
-        role='student'
-    )
+    end_assessment_return = get_end_assessment_return(request.user.anonymous_user_id)
 
-    # These claims are optional. They are necessary to set in order to properly verify the verified_user claim,
-    # if the Proctoring Tool includes it in the JWT.
-    # TODO: This will need to have additional consideration for PII.
-    optional_user_identity_claims = get_optional_user_identity_claims()
-    lti_consumer.set_proctoring_user_data(
-        **optional_user_identity_claims
-    )
-
-    try:
-        lti_response = lti_consumer.check_and_decode_proctoring_token(request.POST.get('JWT'))
-    except (MalformedJwtToken, TokenSignatureExpired):
-        return JsonResponse(
-            {'error': 'invalid_grant'},
-            status=400,
-        )
-    except NoSuitableKeys:
-        return JsonResponse(
-            {'error': 'invalid_client'},
-            status=400
-        )
-    except (BadJwtSignature, InvalidClaimValue, MissingRequiredClaim):
-        # TODO: I'm not sure whether this is the right OIDC error ID.
-        return JsonResponse(
-            {'error': 'invalid_token'},
-            status=400
+    # If end_assessment_return was provided by the Proctoring Tool, and end_assessment was True, then the Assessment
+    # Platform MUST send an End Assessment message to the Proctoring Tool. Otherwise, the Assessment Platform can
+    # complete its normal post-assessment flow.
+    if end_assessment_return:
+        exam_attempt = ExamAttempt.objects.get(pk=attempt_id)
+        exam = exam_attempt.exam
+        lti_config_id = exam.provider.lti_configuration_id
+        lti_config = LtiConfiguration.objects.get(id=lti_config_id)
+
+        proctoring_launch_data = Lti1p3ProctoringLaunchData(
+            # TODO: This is incorrect. It should be an integer starting at 1 and increasing by 1 for each subsequent
```

Comment:
Question: So this still needs to be addressed correct? Just a placeholder I'm assuming?

Reply:
I will replace `exam_attempt.id` with `exam_attempt.attempt_number`, per Zach's comment, which will address this. I'll remove the TODO.